plugins/inputs/sql/sql.go

```diff
@@ -375,16 +387,31 @@ func (s *SQL) Start(_ telegraf.Accumulator) error {
 	s.db.SetConnMaxLifetime(time.Duration(s.MaxLifetime))
 	s.db.SetMaxOpenConns(s.MaxOpenConnections)
 	s.db.SetMaxIdleConns(s.MaxIdleConnections)
+	return nil
+}
 
+func (s *SQL) ping() error {
 	// Test if the connection can be established
 	s.Log.Debug("Testing connectivity...")
 	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(s.Timeout))
-	err = s.db.PingContext(ctx)
+	err := s.db.PingContext(ctx)
 	cancel()
 	if err != nil {
-		return fmt.Errorf("connecting to database failed: %w", err)
+		if s.DisconnectedServersBehavior == "error" {
+			return fmt.Errorf("unable to connect to database: %w", err)
+		}
+		s.Log.Errorf("unable to connect to database: %s", err)
+		return nil
```

Comment:
What do you think about having the disconnected server behavior logic exist in `Start`, instead of `ping`? That way ping is focused only on reaching out to the server.

Additionally, the `prepareStatements` doesn't need to know about serverConnected either as we can skip that logic.

```suggestion
		return fmt.Errorf("unable to connect to database: %w", err)
```